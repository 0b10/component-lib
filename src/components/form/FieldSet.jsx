import React from "react";
import PropTypes from "prop-types";

export const FieldSet = props => (
  <fieldset style={(props.styles && props.styles.fieldset) || styles.fieldset}>
    <legend style={(props.style && props.styles.legend) || styles.legend}>
      {props.legend}
    </legend>
    {props.children}
  </fieldset>
);

// FIXME: determine how to use the global theme.
const styles = {
  fieldset: {
    padding: "50px",
    borderRadius: "4px"
  },
  legend: undefined
};

FieldSet.propTypes = {
  //** The legend text */
  legend: PropTypes.string,
  styles: PropTypes.shape({
    fieldset: PropTypes.object,
    legend: PropTypes.object
  })
};
