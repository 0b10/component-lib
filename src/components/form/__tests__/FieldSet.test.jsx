import React from "react";
import { mount, shallow } from "enzyme";
// Local
import { FieldSet } from "../FieldSet";

describe("Package: form", () => {
  // +++ basic render +++
  describe("Component: FieldSet", () => {
    it("should render something", () => {
      const wrapper = shallow(
        <FieldSet legend="test-legend">
          <div>test</div>
        </FieldSet>
      );
      expect(wrapper.length).toBe(1);
    });

    // +++ legend text +++
    describe.each`
      legend
      ${undefined}
      ${"test-legend"}
    `("The <legend/> tag", ({ legend }) => {
      it(`should render ${
        legend ? legend : "an empty string"
      } (props.legend)`, () => {
        const wrapper = shallow(
          <FieldSet legend={legend}>
            <div>test</div>
          </FieldSet>
        ).find("legend");
        expect(wrapper.text()).toBe(legend ? legend : "");
      });
    });

    // +++ snapshot +++
    it("should match the snapshot", () => {
      const wrapper = shallow(
        <FieldSet legend="test-legend">
          <div>test</div>
        </FieldSet>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
