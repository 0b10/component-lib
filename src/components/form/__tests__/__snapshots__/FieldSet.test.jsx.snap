// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package: form Component: FieldSet should match the snapshot 1`] = `
<fieldset
  style={
    Object {
      "borderRadius": "4px",
      "borderWidth": "1px",
      "padding": "50px",
    }
  }
>
  <legend>
    test-legend
  </legend>
  <div>
    test
  </div>
</fieldset>
`;
